pkg/client/camel/applyconfiguration/core/v1/affinity.go

```diff
@@ -0,0 +1,58 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+// AffinityApplyConfiguration represents an declarative configuration of the Affinity type for use
+// with apply.
+type AffinityApplyConfiguration struct {
```

Comment:
I think this is some issue with the autogen. We should not have kubernetes resources here I guess. @astefanutti fyi

Reply:
I agree it's not ideal to have all these types duplicated. I've giving this another cycle, and it seems there is issues with both embedding and cross-referencing.

@lburgazzoli I know I've been suggesting to embed `corev1.PodCondition`, but maybe it's not needed in the short term so we can look into those issues. WDYT?

Reply:
As an alternative I can replace the struct with an custom one with the same fields and explicit copy of fields.
Both are fine with me, @astefanutti @squakez cast your vote :)

Reply:
If we don't have all those bunch of duplicated types it would be more maintainable, so, I'd prefer that option.

Reply:
One alternative that seems to work is to not inline the `corev1.PodCondition`, i.e. adding a field, and remove `metav1` and `corev1` from the code-generator input dirs. That way the API still has the corresponding info, but the generated apply configurations do no contain any duplicate, and instead refer to the API rather than the apply configurations, which is acceptable IMO.

Reply:
ok, will try that

Reply:
done, please have a look at the latest commits (it is a little bit weird as there is a Pod nested field in the PodCondition  (if you have better names, please suggest)

Reply:
Maybe -re-naming the field to `Readiness` or `Ready` would make it clear it's the ready condition from the Pod that's provided?

Reply:
done